content/docs/iac/guides/building-extending/providers/implementers/go.md

```diff
@@ -0,0 +1,497 @@
+---
+title_tag: "Implement a provider in Go"
+meta_desc: "Build a Pulumi provider in Go using the gRPC bindings directly, without the higher-level SDK."
+title: Implement a provider in Go
+h1: Implement a provider in Go
+meta_image: /images/docs/meta-images/docs-meta.png
+menu:
+    iac:
+        name: Go
+        parent: iac-guides-provider-implementers
+        weight: 30
+---
+
+This guide shows how to implement a Pulumi provider in Go using the gRPC bindings directly, without the [Pulumi Go Provider SDK](/docs/iac/guides/building-extending/providers/sdks/pulumi-go-provider-sdk/). This approach gives you full control over provider behavior and schema definition.
+
+{{% notes type="info" %}}
+For most Go providers, the [Pulumi Go Provider SDK](/docs/iac/guides/building-extending/providers/sdks/pulumi-go-provider-sdk/) provides a more ergonomic experience with schema inference and less boilerplate. Use this direct approach when you need precise control over the protocol or want to understand exactly how providers work.
+{{% /notes %}}
+
+## Prerequisites
+
+You'll need Go 1.21 or later, and a basic understanding of the [provider protocol](/docs/iac/guides/building-extending/providers/implementers/protocol-reference/).
```

Comment:
Nit: Go 1.24 is currently the lowest supported version of Go, so not sure why we'd mention 1.21.